src/agora/dobeamweights.cc

```diff
@@ -85,17 +85,77 @@ DoBeamWeights::~DoBeamWeights() {
   std::free(calib_gather_buffer_);
 }
 
-EventData DoBeamWeights::Launch(size_t tag) {
-  ComputeFullCsiBeams(tag);
+EventData DoBeamWeights::Launch(size_t tag) {  // Compute Full-CSI Beams
+  const size_t frame_id = gen_tag_t(tag).frame_id_;
```

Comment:
I would put the function call back for now.    Harder to track changes inside the function